HealthcareApp/src/Healthcare.Application/Implementations/AuthenticationService.cs

```diff
@@ -49,6 +54,9 @@ public async Task<IdentityResult> RegisterAsync(RegisterUserDTO registerUserDto)
 
         await _userManager.AddToRoleAsync(user, registerUserDto.Role);
 
+        if (registerUserDto.Role == "Doctor")
+           await _scheduleRepository.CreateDefaultWorkingScheduleAsync(user.Id);
+
```

Comment:
If you decided to use MediatR, I recommend you to use notifications instead of locating this code here. You can raise event (notification) about user registration, and in some notification handler create working schedule for your doctor.
This way, if you get new roles and your logic grows, your register method won't grow and you will follow OCP.
Sample of notifications in mediatr - https://medium.com/codenx/mediatr-with-notification-system-use-case-in-asp-net-core-21fce433bb19